src/qt/bitcoin.cpp

```diff
@@ -393,6 +394,7 @@ WId BitcoinApplication::getMainWinId() const
 static void SetupUIArgs()
 {
     gArgs.AddArg("-choosedatadir", strprintf("Choose data directory on startup (default: %u)", DEFAULT_CHOOSE_DATADIR), ArgsManager::ALLOW_ANY, OptionsCategory::GUI);
+    gArgs.AddArg("-guisettings=<path>", "Choose a custom data directory especially for the Qt Settings", ArgsManager::ALLOW_ANY, OptionsCategory::GUI);
```

Comment:
Note that due to initialization order constraints, this option can only be provided on the command line.
It won't work in the configuration file because at that time, QSettings was already parsed.

Reply:
I think you are wrong.
Take a look at `src/qt/bitcoin.cpp` line 451.
There you'll see that `SetupUIArgs()` is executed before the PR feature.